x-pack/plugins/cases/public/components/user_actions/assignees.tsx

```diff
@@ -5,13 +5,159 @@
+      <>
+        {children}
+        {','}
+      </>
+    );
+  }
+
+  return children;
+};
+FormatListItem.displayName = 'FormatListItem';
+
+const shouldAddComma = (index: number, arrayLength: number) => {
+  return arrayLength > 0 && index !== arrayLength - 1;
+};
+
+const shouldAddAnd = (index: number, arrayLength: number) => {
+  return arrayLength > 1 && index === arrayLength - 1;
+};
+
+const Themselves: React.FC<{
+  index: number;
+  numOfAssigness: number;
+}> = ({ index, numOfAssigness }) => (
+  <FormatListItem index={index} listSize={numOfAssigness}>
+    <>{i18n.THEMSELVES}</>
+  </FormatListItem>
+);
+Themselves.displayName = 'Themselves';
+
+const AssigneeComponent: React.FC<{
+  assignee: Assignee;
+  index: number;
+  numOfAssigness: number;
+}> = ({ assignee, index, numOfAssigness }) => (
+  <FormatListItem index={index} listSize={numOfAssigness}>
+    <UserToolTip profile={assignee.profile}>
+      <strong>{getName(assignee.profile?.user)}</strong>
+    </UserToolTip>
+  </FormatListItem>
+);
+AssigneeComponent.displayName = 'Assignee';
+
+interface AssigneesProps {
+  assignees: Assignee[];
+  currentUserProfile?: UserProfileWithAvatar;
+}
+
+const AssigneesComponent = ({ assignees, currentUserProfile }: AssigneesProps) => (
+  <>
+    {assignees.length > 0 && (
+      <EuiFlexGroup alignItems="center" gutterSize="xs">
+        {assignees.map((assignee, index) => {
+          const usernameDataTestSubj = getUsernameDataTestSubj(assignee);
+
+          return (
+            <EuiFlexItem
+              data-test-subj={`ua-assignee-${usernameDataTestSubj}`}
+              grow={false}
+              key={assignee.uid}
+            >
+              <EuiText size="s" className="eui-textBreakWord">
+                {isCurrentUser(assignee, currentUserProfile) ? (
+                  <Themselves index={index} numOfAssigness={assignees.length} />
+                ) : (
+                  <AssigneeComponent
+                    assignee={assignee}
+                    index={index}
+                    numOfAssigness={assignees.length}
+                  />
+                )}
+              </EuiText>
+            </EuiFlexItem>
+          );
+        })}
+      </EuiFlexGroup>
+    )}
+  </>
+);
+AssigneesComponent.displayName = 'Assignees';
+const Assignees = memo(AssigneesComponent);
+
+const isCurrentUser = (assignee: Assignee, currentUserProfile?: UserProfileWithAvatar) => {
+  return assignee.uid === currentUserProfile?.uid;
+};
+
+const getLabelTitle = (
+  userAction: UserActionResponse<AssigneesUserAction>,
+  userProfiles?: Map<string, UserProfileWithAvatar>,
+  currentUserProfile?: UserProfileWithAvatar
+) => {
+  const assignees = userAction.payload.assignees.map((assignee) => {
+    const profile = userProfiles?.get(assignee.uid);
+    return {
+      uid: assignee.uid,
+      profile,
+    };
+  });
+
+  return (
+    <EuiFlexGroup alignItems="baseline" gutterSize="xs" component="span" responsive={false}>
```

Comment:
Just checking this is the desired behavior - the first part of the label uses the user ID, and the rest uses the full names. 

<img width="706" alt="image" src="https://user-images.githubusercontent.com/7405507/186675101-2ae30688-81e7-4df4-bf37-705177554460.png">

GitHub by comparison only uses user IDs:

<img width="550" alt="image" src="https://user-images.githubusercontent.com/7405507/186675171-732a709e-35e3-4e14-abd8-25c200aeb720.png">

Reply:
Yeah the mocks are showing the `display_name`, we don't have that within elasticsearch yet so the fallback order is `full name -> email -> username`

Once we have support for display name the user can customize the name that is displayed.